apps/judicial-system/web/src/routes/Court/Indictments/Conclusion/Conclusion.tsx

```diff
@@ -80,76 +81,202 @@ const Conclusion: React.FC = () => {
+
+      router.push(destination)
+    },
+    [setAndSendCaseToServer, setWorkingCase, transitionCase, workingCase],
+  )
+
+  const handleCompletion = useCallback(
+    async (destination: string) => {
+      const success = await setAndSendCaseToServer(
+        [
+          {
+            indictmentRulingDecision: selectedDecision,
+            indictmentDecision: IndictmentDecision.COMPLETING,
+            force: true,
+          },
+        ],
+        workingCase,
+        setWorkingCase,
+      )
+
+      if (!success) {
+        return
+      }
+
+      router.push(`${destination}/${workingCase.id}`)
+    },
+    [selectedDecision, setAndSendCaseToServer, setWorkingCase, workingCase],
+  )
+
+  const handlePostponementUntilVerdict = useCallback(
+    async (destination: string) => {
+      const updates = {
+        ...(workingCase.indictmentDecision !==
+          IndictmentDecision.POSTPONING_UNTIL_VERDICT && {
+          indictmentDecision: IndictmentDecision.POSTPONING_UNTIL_VERDICT,
+        }),
+        ...((postponement?.isSettingVerdictDate || workingCase.courtDate) && {
+          courtDate:
+            postponement?.isSettingVerdictDate && courtDate?.date
+              ? {
+                  date: formatDateForServer(new Date(courtDate.date)),
+                  location: courtDate.location,
+                }
+              : null,
+        }),
+      }
+
+      const success =
+        Object.keys(updates).length > 0
+          ? await setAndSendCaseToServer(
+              [
+                {
+                  ...updates,
+                  force: true,
+                },
+              ],
+              workingCase,
+              setWorkingCase,
+            )
+          : true
+
+      if (!success) {
+        return
+      }
+
+      router.push(`${destination}/${workingCase.id}`)
+    },
+    [
+      courtDate?.date,
+      courtDate?.location,
+      postponement?.isSettingVerdictDate,
+      setAndSendCaseToServer,
+      setWorkingCase,
       workingCase,
+    ],
+  )
+
+  const handlePostponementIndefinitely = useCallback(
+    async (destination: string) => {
+      const updateSuccess = await setAndSendCaseToServer(
+        [
+          {
+            courtDate: null,
+            postponedIndefinitelyExplanation: postponement?.reason,
+          },
+        ],
+        workingCase,
+        setWorkingCase,
+      )
+
+      if (!updateSuccess) {
+        return
+      }
+
+      router.push(`${destination}/${workingCase.id}`)
+    },
+    [postponement?.reason, setAndSendCaseToServer, setWorkingCase, workingCase],
+  )
+
+  const handlePostponement = useCallback(
+    async (destination: string) => {
+      const updateCourtDateSuccess = await sendCourtDateToServer([
+        { postponedIndefinitelyExplanation: null, force: true },
+      ])
+
+      const updateSuccess = await setAndSendCaseToServer(
+        [
+          {
+            indictmentDecision: IndictmentDecision.POSTPONING,
+          },
```

Comment:
Should the selected courtDate not be set here?

Reply:
It was being set above this code, in another function `sendCourtDateToServer`. I've now combined this into a single call to the server. That makes this function a bit cleaner also..